extensions/common/vault/vault-aws/src/main/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultExtension.java

```diff
@@ -36,6 +39,9 @@ public class AwsSecretsManagerVaultExtension implements ServiceExtension {
     @Setting
     private static final String VAULT_AWS_REGION = "edc.vault.aws.region";
 
+    @Setting
+    private static final String AWS_ENDPOINT_OVERRIDE = "edc.aws.endpoint.override";
```

Comment:
I think this setting should be different than the one used to create the S3 client, and be called `edc.vault.aws.endpoint.override`, to align with the region setting.

plus, settings can be injected in a different way, [as described in this DR](https://github.com/eclipse-edc/Connector/tree/4819c8ec3036654391e4b2f81b414667797e3e48/docs/developer/decision-records/2024-11-06-configuration-injection)

Please adapt and add some `description` to the annotation